lib/std/zig/render.zig

```diff
@@ -1081,51 +1061,37 @@ fn renderWhile(gpa: *Allocator, ais: *Ais, tree: ast.Tree, while_node: ast.full.
     }
 
     const rparen = tree.lastToken(while_node.ast.cond_expr) + 1;
+    const first_then_token = tree.firstToken(while_node.ast.then_expr);
     const last_then_token = tree.lastToken(while_node.ast.then_expr);
     const src_has_newline = !tree.tokensOnSameLine(rparen, last_then_token);
 
     if (src_has_newline) {
+        const base_node_is_if = mem.eql(u8, tree.tokenSlice(while_node.ast.while_token), "if");
+        const newline_before_then_token = !tree.tokensOnSameLine(rparen, first_then_token);
+        const space_before_then_token: Space = if (newline_before_then_token or base_node_is_if) .newline else .space;
+        const indent_expression = base_node_is_if or !nodeIsIf(then_tag) or newline_before_then_token;
```

Comment:
This all feels a bit icky to me, especially comparing against the 'if' string. I couldn't find a way to get the original type of the `while_node`. Of course we could add a 'is_if_node' param, but at that point I question using `renderWhile()` for rending 'if's at all :)

Reply:
ifreund has suggested on IRC that the `base_node_is_if` could just be removed, and this sounds sensible to me. Note that this would result in a change in behaviour - the following would no longer be reformatted onto 3 lines:
```zig
if (a) |x| if (x)
    f(x);
```